src/resolver.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  ABIChanged as ABIChangedEvent,
+  AddrChanged as AddrChangedEvent,
+  AuthorisationChanged as AuthorisationChangedEvent,
+  ContenthashChanged as ContenthashChangedEvent,
+  InterfaceChanged as InterfaceChangedEvent,
+  NameChanged as NameChangedEvent,
+  PubkeyChanged as PubkeyChangedEvent,
+  TextChanged as TextChangedEvent
+} from './types/Resolver/Resolver'
+
+import {
+  AddrChanged,
+  NameChanged,
+  AbiChanged,
+  PubkeyChanged,
+  TextChanged,
+  ContenthashChanged,
+  InterfaceChanged,
+  AuthorisationChanged,
+} from './types/schema'
+
+export function handleAddrChanged(event: AddrChangedEvent): void {
+  let id = event.block.number.toString().concat('-').concat(event.logIndex.toString())
+  let resolverEvent = new AddrChanged(id)
+  resolverEvent.resolver = event.address
+  resolverEvent.node = event.params.node
+  resolverEvent.resolverID = event.address.toHexString().concat('-').concat(event.params.node.toHexString())
+  resolverEvent.address = event.params.a
+  resolverEvent.save()
+}
+
+
+export function handleNameChanged(event: NameChangedEvent): void {
```

Comment:
There's a lot of repeated code here; could it be abstracted to a common method?

Reply:
yes good idea